client-cli/src/main/java/org/eclipse/dataspaceconnector/registration/cli/RegistrationServiceCli.java

```diff
@@ -14,24 +14,43 @@
 
 package org.eclipse.dataspaceconnector.registration.cli;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.OkHttpClient;
+import org.eclipse.dataspaceconnector.iam.did.web.resolution.WebDidResolver;
 import org.eclipse.dataspaceconnector.registration.client.api.RegistryApi;
+import org.eclipse.dataspaceconnector.spi.monitor.ConsoleMonitor;
+import org.jetbrains.annotations.NotNull;
 import picocli.CommandLine;
 import picocli.CommandLine.Command;
 
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.Boolean.parseBoolean;
+import static org.eclipse.dataspaceconnector.common.configuration.ConfigurationFunctions.propOrEnv;
+import static org.eclipse.dataspaceconnector.registration.cli.ClientUtils.createApiClient;
 
 @Command(name = "registration-service-cli", mixinStandardHelpOptions = true,
         description = "Client utility for MVD registration service.",
         subcommands = {
                 ParticipantsCommand.class
         })
 public class RegistrationServiceCli {
-    @CommandLine.Option(names = "-s", required = true, description = "Registration service URL", defaultValue = "http://localhost:8182/authority")
+
+    private static final ObjectMapper MAPPER = new ObjectMapper().disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+    private static final String USE_HTTPS_SCHEME = "did.web.use.https";
+
+    @Deprecated
+    @CommandLine.Option(names = "-s", description = "Registration service URL. Deprecated. Use -d instead.", defaultValue = "http://localhost:8182/authority")
     String service;
 
-    @CommandLine.Option(names = "-d", required = true, description = "Client DID")
+    @CommandLine.Option(names = { "-d", "--dataspace-did"}, required = false, description = "Dataspace Authority DID.", defaultValue = "")
+    String dataspaceDid;
+
+    @CommandLine.Option(names = { "-c", "--client-did"}, required = true, description = "Client DID.")
     String clientDid;
 
     @CommandLine.Option(names = "-k", required = true, description = "File containing the private key in PEM format")
```

Comment:
Provide long name variant for this one too?

Reply:
done